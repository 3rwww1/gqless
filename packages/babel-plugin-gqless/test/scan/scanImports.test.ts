import { scanImports } from './setup'

it('ignores unused imports', () => {
  expect(
    scanImports(`
      import * as a from 'mod'
      import b from 'mod'
      import c from 'mod'
      const d = require('mod')
    `)
  ).toMatchSnapshot()
})

describe('modules >', () => {
  test('namespace imports', () => {
    expect(
      scanImports(`
        import * as a from 'mod'
        a.b
      `)
    ).toMatchSnapshot()
  })

  test('default imports', () => {
    expect(
      scanImports(`
        import a from 'mod'
        a
      `)
    ).toMatchSnapshot()
  })

  test('named imports', () => {
    expect(
      scanImports(`
        import { a } from 'mod'
        a
      `)
    ).toMatchSnapshot()
  })
})

describe('commonjs >', () => {
  test('require destructure', () => {
    expect(
      scanImports(`
        const { a } = require('mod')
      `)
    ).toMatchSnapshot()
  })

  test('require property', () => {
    expect(
      scanImports(`
        const a = require('mod').a
      `)
    ).toMatchSnapshot()
  })
})
