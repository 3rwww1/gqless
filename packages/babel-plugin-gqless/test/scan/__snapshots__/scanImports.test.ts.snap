// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commonjs > require destructure 1`] = `
Object {
  "mod": Array [
    "a",
  ],
}
`;

exports[`commonjs > require property 1`] = `
Object {
  "mod": Array [
    "a",
  ],
}
`;

exports[`ignores unused imports 1`] = `
Object {
  "mod": Array [],
}
`;

exports[`modules > default imports 1`] = `
Object {
  "mod": Array [
    "default",
  ],
}
`;

exports[`modules > named imports 1`] = `
Object {
  "mod": Array [
    "a",
  ],
}
`;

exports[`modules > namespace imports 1`] = `
Object {
  "mod": Array [
    "b",
  ],
}
`;
